sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaReadSchemaTransformProvider.java

```diff
@@ -262,17 +270,22 @@ public static class ErrorFn extends DoFn<byte[], Row> {
     private SerializableFunction<byte[], Row> valueMapper;
     private Counter errorCounter;
     private Long errorsInBundle = 0L;
+    private final boolean failOnError;
 
-    public ErrorFn(String name, SerializableFunction<byte[], Row> valueMapper) {
+    public ErrorFn(String name, SerializableFunction<byte[], Row> valueMapper, boolean failOnError) {
       this.errorCounter = Metrics.counter(KafkaReadSchemaTransformProvider.class, name);
       this.valueMapper = valueMapper;
+      this.failOnError = failOnError;
     }
 
     @ProcessElement
     public void process(@DoFn.Element byte[] msg, MultiOutputReceiver receiver) {
       try {
         receiver.get(OUTPUT_TAG).output(valueMapper.apply(msg));
```

Comment:
Best practice is to not catch errors from `Reciever.output(...)` which may involve arbitrary downstream fused operations. Instead, catch only the errors from `valueMapper.apply(msg)` and output outside of the try block.

Reply:
Done.